### PR TITLE
feat: add `fisher` completion spec

### PR DESCRIPTION
**Pull request by Tim4497**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Missing support for [Fisher](https://github.com/jorgebucaran/fisher) autocomplete.

**What is the new behavior (if this is a feature change)?**

Added basic autocomplete, there is no complete list of all available fisher plugins so this will be a work in progress.

### CODE REVIEW

File: src/fisher.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    {
+      name: "list",
+      description: "List plugins",
+      args: [
+        {},
+        {
+          name: "RegEx",
+          description: "Search in list with regular expression",
+          isOptional: true,
+        },
```
Comment (mschrage) [inline src/fisher.ts:130]: Is it intentional to have an empty argument here?

Is this how the command would be invoked?
```
fisher list xxx '.*?'
```

Comment (Tim4497) [reply src/fisher.ts:130]: My intention was to indicate that you just can press enter to list all plugins, (like a empty regex)

Comment (fedeci) [reply src/fisher.ts:130]: @Tim4497 then you can safely remove that empty arg!
With this structure:
```ts
args: [
        {
          name: "RegEx",
          description: "Search in list with regular expression",
          isOptional: true,
        },
]
```

- `fisher list`: returns a list of all plugins
- `fisher list [regex]` will return all the plugins that match the regex

**Comment by Tim4497** [comment]
worked on another completion spec and merged it into master but now its reverted again

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**